debian/sourcemapper.go

```diff
@@ -142,16 +143,19 @@ func (m *sourcesMap) fetchSources(ctx context.Context, url string) error {
 		}
 		binaries := hdr.Get("Binary")
 		m.mu.Lock()
-		if m.sourceMap[source] == nil {
-			m.sourceMap[source] = make(map[string]struct{})
+		if m.sourceMap[distro] == nil {
+			m.sourceMap[distro] = make(map[string]map[string]struct{})
+		}
+		if m.sourceMap[distro][source] == nil {
+			m.sourceMap[distro][source] = make(map[string]struct{})
 		}
 		for _, bin := range strings.Split(binaries, ", ") {
-			m.sourceMap[source][bin] = struct{}{}
+			m.sourceMap[distro][source][bin] = struct{}{}
 		}
 		m.mu.Unlock()
 	}
-	switch {
-	case errors.Is(err, io.EOF):
+	switch err {
+	case io.EOF:
```

Comment:
This shouldn't be rewritten, because it means that the error can't be wrapped.

Reply:
Guess it's not hurting anyone. I'll revert